pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMessageDecoder.java

```diff
@@ -46,9 +47,25 @@ public interface StreamMessageDecoder<T> {
    * @param topicName Topic name of the stream
    * @throws Exception If an error occurs
    */
+  @Deprecated
```

Comment:
Since we are deprecating it, add a default implementation which throws `UnsupportedOperationException`, then we can only choose to implement one of the `init()` and clean up the decoders maintained within the code base